File: Histogram Ugliness/Histogram_Ugliness.py
def Histogram_Ugliness():

    total_bars = int(input())
    bars = list(map(int,input().split()))
    ugli = 0
    last = 0

    for i in range(total_bars):
        if bars[i] > last:
            ugli += bars[i] - last 
        last = bars[i]
        if i == total_bars - 1:
            ugli += bars[i]
            break
        if bars[i+1] < bars[i]:
            ugli += bars[i] - bars[i+1]
    if total_bars == 1:
        ugli -= bars[0]
        return ugli

    for i in range(total_bars):
        if i == 0:
            if bars[i+1] < bars[i]:
                ugli -= bars[i] - bars[i+1]
                bars[i] = bars[i+1]
            
        if i == total_bars-1:
            if bars[i-1] < bars[i]:
                ugli -= bars[i] - bars[i-1]
                bars[i] = bars[i-1]
            
        else:
            if bars[i] > bars[i-1] and bars[i] >  bars[i+1]:
                if bars[i-1] > bars[i+1]:
                    ugli -= (bars[i] - bars[i-1])
                    bars[i] = bars[i-1]
                else:
                    ugli -= (bars[i] - bars[i+1])
                    bars[i] = bars[i+1]
                
            
            
    return ugli  
            
        
remained_test_cases = 1 
remained_test_cases = int(input())

while remained_test_cases  > 0:
    print(Histogram_Ugliness())
    remained_test_cases -= 1 
    

File: Histogram Ugliness/Histogram_Ugliness.cpp
#include <bits/stdc++.h>
#define ll long long
using namespace std;


ll Histogram_Ugliness()
{ ll total_bars;
  cin>>total_bars;
  vector<ll> bars;
  
  
  
  for(ll i = 0;i < total_bars;i++)
  {   ll height;
      cin>>height;
      bars.push_back(height);
  }
  if (total_bars == 1)
  return(bars[0]);
  
  ll ugliness = 0;
  ll last = 0;     //previous height
  for(ll i = 0;i < total_bars;i++)
  {
      if(bars[i] > last)
      {  ugliness += bars[i] - last;
         
      }
      
      if (i == total_bars - 1)
      {ugliness += bars[i];
      break;}
      
      if (bars[i+1] < bars[i])
       ugliness += bars[i] - bars[i+1];
       
       last = bars[i];
       
  }
  
  for(ll  i = 0;i< total_bars;i++)
  {  if(i == 0)
     {
         if(bars[i+1] < bars[i])
         {
             ugliness -= bars[i] - bars[i+1];
             bars[i] = bars[i+1];
         }
     }
     else if(i == total_bars - 1 )
     {
         if(bars[i] > bars[i-1])
         {
             ugliness -= bars[i] - bars[i-1];
             bars[i] = bars[i-1];
         }
     }
     else
     {
         if(bars[i] > bars[i-1] && bars[i] > bars[i+1])
         {
             ugliness -= bars[i] - max(bars[i+1],bars[i-1]);
             bars[i] = max(bars[i-1],bars[i+1]);
         }
     }
      
  }
  return (ugliness);
  

}


int main() {
	int remained_test_cases = 1 ;
	cin>>remained_test_cases;
	while(remained_test_cases > 0)
	{   cout<<Histogram_Ugliness()<<"\n";
	    remained_test_cases--;
	}
	return 0;
}
